### DISCUSSION TITLE
--flannel-iface not working

### DISCUSSION THREAD

**Discussion by mcarbonneaux** [k3s-io/k3s#11341] [Q&A]
**Environmental Info:**
K3s Version: 
```
# k3s -v
k3s version v1.30.6+k3s1 (1829eaae)
go version go1.22.8
```

Node(s) CPU architecture, OS, and Version: 
```
# uname -a
Linux anakai.ch2o.info 6.8.0-48-generic #48-Ubuntu SMP PREEMPT_DYNAMIC Fri Sep 27 14:04:52 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

Cluster Configuration:
single node, docker mode, public interface, and private dummy interface.

**Describe the bug:**
i try to not expose on public interface anything.
i found a way to do it for quasi everything... but not for flannel vxlan listen port that listen on any interfaces : `0.0.0.0:8472`..
i tryed to use `--flannel-iface` but it continue to listen to any...

my private interface is kubelet0 and is linux dummy interface, the idea is to force to listen all port that not need to be exposed to public interface on local dummy interface (i've tryed loopback but many part of k8s not whant to listen on it).

**Steps To Reproduce:**
NODE_IP=<kubelet0 ip: 192.168.0.1> 
PUBLIC_IP< the public ip of my server>

```
export INSTALL_K3S_EXEC="server \
    --advertise-address=${NODE_IP} \
    --bind-address=${NODE_IP} \
    --node-ip=${NODE_IP} \
    --flannel-iface=kubelet0 \
    --node-external-ip=${PUBLIC_IP} \
    --debug \
    -v 100 \
    --docker \
    --data-dir=/data/k3s \
    --server=${NODE_IP} \
    --tls-san=${NODE_IP} \
    --kubelet-arg=node-ip=${NODE_IP} \
    --kubelet-arg=address=${NODE_IP} "

export K3S_URL=https://${NODE_IP}:6443
curl -sfL https://get.k3s.io | sh -s - --docker
```

**Expected behavior:**
that flannel listen on NODE_IP on my dummy interface (192.168.0.1).

**Actual behavior:**
listen on any interface:
```
# netstat -lnput | grep 8472
udp        0      0 0.0.0.0:8472            0.0.0.0:*                           -
```

**Comment by brandond** [comment]
I don't think flannel actually supports binding to a specific IP like that. I would take this question to https://github.com/flannel-io/flannel
